src/core/SpriteAnimator.tsx

```diff
@@ -1,72 +1,176 @@
+   * @default true
+   */
   asSprite?: boolean
+  /** Allows for manual update of the sprite animation e.g: via ScrollControls.
+   * Value should be between 0 and 1
+   */
   offset?: number
+  /** Allows the sprite animation to start from the end towards the start */
   playBackwards?: boolean
+  /** Allows the animation to be paused after it ended so it can be restarted on demand via autoPlay */
   resetOnEnd?: boolean
+  /** Array of Vector3-like positions for creating multiple instances of the sprite */
+  instanceItems?: (THREE.Vector3 | [number, number, number])[]
+  /** The maximum number of instances to render (for buffer size calculation)
+   * @default 1
+   */
   maxItems?: number
-  instanceItems?: any[]
-  spriteDataset?: any
+  /** Pre-parsed sprite data, usually from useSpriteLoader ready for use */
+  spriteDataset?: {
+    spriteTexture: THREE.Texture
+    spriteData: SpriteData
+  }
+  /** Configuration options for the canvas context when loading textures
+   * @see https://developer.mozilla.org/en-US/docs/Web/API/CanvasRenderingContext2D/CanvasRenderingContext2D
+   */
   canvasRenderingContext2DSettings?: CanvasRenderingContext2DSettings
+  /** Controls whether frame positions are rounded for precise pixel alignment.
+   * Enable this if you notice slight texture bleeding between frames.
+   * @default false
+   */
   roundFramePosition?: boolean
-  /** Additional properties to be passed to the underlying mesh/instance
+  /** Additional properties to be passed to both simple mesh and instance components.
    * @example { frustumCulled: false, renderOrder: 1 }
+   * @see https://threejs.org/docs/#api/en/core/Object3D
    */
   meshProps?: CommonMeshProps
 } & JSX.IntrinsicElements['group']
 
 type SpriteAnimatorState = {
-  /** The user-defined, mutable, current goal position along the curve, it may be >1 or <0 */
   current: number | undefined
-  /** The 0-1 normalised and damped current goal position along curve */
   offset: number | undefined
-  hasEnded: boolean | undefined
-  ref: React.MutableRefObject<any> | undefined | null | ((instance: any) => void)
+  imageUrl?: string
+  reset: () => void
+  hasEnded: boolean
+  ref: React.RefObject<THREE.Group> | null | ((instance: THREE.Group) => void)
 }
 
 type Scale = number | [number, number, number] | Vector3
 
 const context = React.createContext<SpriteAnimatorState | null>(null)
 
-export function useSpriteAnimator() {
+export function useSpriteAnimator(): SpriteAnimatorState | null {
   return React.useContext(context)
 }
 
+// Type guard for SpriteData
+function isSpriteData(data: SpriteData | null): data is SpriteData {
+  return data !== null && 'meta' in data && 'frames' in data
+}
+
 const geometry = new THREE.PlaneGeometry(1, 1)
 
-export const SpriteAnimator: React.FC<SpriteAnimatorProps> = /* @__PURE__ */ React.forwardRef(
+export const SpriteAnimator: React.ForwardRefExoticComponent<SpriteAnimatorProps & React.RefAttributes<THREE.Group>> = React.forwardRef(
```

Comment:
type casting is not the greatest, you can let TS infer using `forwardRef` generics `<Ref, Props>`:

```ts
const SpriteAnimator = forwardRef<Ref, Props>((props, fref) => {...})
```

see: https://github.com/pmndrs/drei/blob/master/src/core/Example.tsx#L28